bpf/lib/static_data.h

```diff
@@ -28,14 +28,19 @@
 
 #define DEFINE_IPV6(NAME, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16) \
 DEFINE_U64_I(NAME, 1) = bpf_cpu_to_be64( \
-			(__u64)(a1) << 56 | (__u64)(a2) << 48 | (__u64)(a3) << 40 | \
-			(__u64)(a4) << 32 | (a5) << 24 | (a6) << 16 | (a7) << 8 | (a8)); \
+			(__u64)(__u8)(a1) << 56 | (__u64)(__u8)(a2) << 48 | \
+			(__u64)(__u8)(a3) << 40 | (__u64)(__u8)(a4) << 32 | \
+			(__u64)(__u8)(a5) << 24 | (__u64)(__u8)(a6) << 16 | \
+			(__u64)(__u8)(a7) << 8  | (__u64)(__u8)(a8)); \
```

Comment:
Is the double cast `(__u64)(__u8)` indeed needed here? Meaning, single one `(__u64)` is not enough?

Reply:
We don't want any values to spill over into the higher bits, so it's okay to be defensive here.